lib/wordmove/environments_list.rb

```diff
@@ -0,0 +1,68 @@
+module Wordmove
+  class EnvironmentsList
+    attr_reader :movefile, :logger, :remote_vhosts, :local_vhost
+
+    class << self
+      def print(cli_options)
+        new(cli_options).print
+      end
+    end
+
+    def initialize(options)
+      @logger = Logger.new(STDOUT).tap { |l| l.level = Logger::INFO }
+      @movefile = Wordmove::Movefile.new(options[:config])
+      @remote_vhosts = []
+      @local_vhost = []
+    end
+
+    def print
+      contents = parse_movefile(movefile: movefile)
+      generate_vhost_list(contents: contents)
+      output
+    end
+
+    private
+
+    def select_vhost(contents:)
+      target = contents.select { |_key, env| env[:vhost].present? }
+      target.map { |key, env| { env: key, vhost: env[:vhost] } }
+    end
+
+    def parse_movefile(movefile:)
+      movefile.fetch
+    end
+
+    def output
+      logger.task('Listing Local')
+      puts output_string(vhost_list: local_vhost)
```

Comment:
I'm so sorry to be late on noticing this...

Why have you opted for a plain `puts` instead of using one of `logger`'s public methods? Is because we don't have a `.plain` and you didn't want to have icons prepended? If that's the case: why do not add a `.plain` method to the logger?

I ask because having a custom logger inside the project should force developers to use it to log output.

I've also just worked on the logger in https://github.com/welaika/wordmove/pull/550: that PR won't involve your new code at all, but you can see that I'm adding logic to the logger, so I'd like to have it to rule all the prints thus gaining a single point of intervention would it be needed in the future.

Lemme know

Reply:
Thanks again!
The # 550 feature is so good that I would like to change the logging implementation accordingly.
Please take some time to think about and correct this PR.

Reply:
> so I'd like to have it to rule all the prints thus gaining a single point of intervention would it be needed in the future.

I understand this requirement. 

Let me confirm if this diff meets your requirements or not.

```diff

diff --git a/lib/wordmove/cli.rb b/lib/wordmove/cli.rb
index fe382d5..8b03f22 100644
--- a/lib/wordmove/cli.rb
+++ b/lib/wordmove/cli.rb
@@ -65,10 +65,10 @@ module Wordmove
     def list
       Wordmove::EnvironmentsList.print(options)
     rescue Wordmove::MovefileNotFound => e
-      puts e.message
+      logger.error(e.message)
       exit 1
     rescue Psych::SyntaxError => e
-      puts "Your movefile is not parsable due to a syntax error: #{e.message}"
+      logger.error("Your movefile is not parsable due to a syntax error: #{e.message}")
       exit 1
     end

diff --git a/lib/wordmove/environments_list.rb b/lib/wordmove/environments_list.rb
index c00ea79..6d99ed1 100644
--- a/lib/wordmove/environments_list.rb
+++ b/lib/wordmove/environments_list.rb
@@ -34,10 +34,10 @@ module Wordmove

     def output
       logger.task('Listing Local')
-      puts output_string(vhost_list: local_vhost)
+      logger.plain(output_string(vhost_list: local_vhost))

       logger.task('Listing Remotes')
-      puts output_string(vhost_list: remote_vhosts)
+      logger.plain(output_string(vhost_list: remote_vhosts))
     end

     def output_string(vhost_list:)
diff --git a/lib/wordmove/logger.rb b/lib/wordmove/logger.rb
index dfc8943..f6bad97 100644
--- a/lib/wordmove/logger.rb
+++ b/lib/wordmove/logger.rb
@@ -38,6 +38,10 @@ module Wordmove
       puts "    ℹ️  info".yellow + " | ".black + message.to_s
     end

+    def plain(message)
+      puts message.to_s
+    end
+
     private

     def padding_length(line)
```

If you hope to release #550 first, I'm going to merge this important logging feature into my pull request. 
Then, I'll make some changes and push them into my pull request by myself 😃 

Thanks for your time and consideration.

Reply:
The diff is absolutely ok and aligned with my intention. I'll merge your feature ASAP, then I'll rebase my #550 on top of it and I'll put in the new implementation of logger methods: you don't have to worry about the context from within this PR (more than you've already done!)

Reply:
@pioneerskies 
I've merged commit: b7fbc23 into my pull request branch. Thanks a lot!